ignite/metrics/regression/kendall_correlation.py

```diff
@@ -0,0 +1,115 @@
+    r"""Calculates the
+    `Kendall rank correlation coefficient <https://en.wikipedia.org/wiki/Kendall_rank_correlation_coefficient>`_.
+
+    .. math::
+        \tau = 1-\frac{2(\text{number of discordant pairs})}{\left( \begin{array}{c}n\\2\end{array} \right)}
+
+    Two prediction-target pairs :math:`(P_i, A_i)` and :math:`(P_j, A_j)`, where :math:`i<j`,
+    are said to be concordant when both :math:`P_i<P_j` and :math:`A_i<A_j` holds
+    or both :math:`P_i>P_j` and :math:`A_i>A_j`.
+
+    The ``number of discordant pairs`` counts the number of pairs that are not concordant.
+
+    The computation of this metric is implemented with
+    `scipy.stats.kendalltau <https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.kendalltau.html>`_.
+
+    - ``update`` must receive output of the form ``(y_pred, y)`` or ``{'y_pred': y_pred, 'y': y}``.
+    - `y` and `y_pred` must be of same shape `(N, )` or `(N, 1)`.
+
+    Parameters are inherited from ``Metric.__init__``.
+
+    Args:
+        variant: variant of kendall rank correlation. ``b`` or ``c`` is accepted.
+            Details can be found
+            `here <https://en.wikipedia.org/wiki/Kendall_rank_correlation_coefficient#Accounting_for_ties>`_.
+            Default: ``b``
+        output_transform: a callable that is used to transform the
+            :class:`~ignite.engine.engine.Engine`'s ``process_function``'s output into the
+            form expected by the metric. This can be useful if, for example, you have a multi-output model and
+            you want to compute the metric with respect to one of the outputs.
+            By default, metrics require the output as ``(y_pred, y)`` or ``{'y_pred': y_pred, 'y': y}``.
+        device: specifies which device updates are accumulated on. Setting the
+            metric's device to be the same as your ``update`` arguments ensures the ``update`` method is
+            non-blocking. By default, CPU.
+
+    Examples:
+        To use with ``Engine`` and ``process_function``, simply attach the metric instance to the engine.
+        The output of the engine's ``process_function`` needs to be in format of
+        ``(y_pred, y)`` or ``{'y_pred': y_pred, 'y': y, ...}``.
+
+        .. include:: defaults.rst
+            :start-after: :orphan:
+
+        .. testcode::
+
+            metric = KendallRankCorrelation()
+            metric.attach(default_evaluator, 'kendall_tau')
+            y_true = torch.tensor([0., 1., 2., 3., 4., 5.])
+            y_pred = torch.tensor([0.5, 2.8, 1.9, 1.3, 6.0, 4.1])
+            state = default_evaluator.run([[y_pred, y_true]])
+            print(state.metrics['kendall_tau'])
+
+        .. testoutput::
+
+            0.4666666666666666
+    """
+
+    def __init__(
+        self,
+        variant: str = "b",
+        output_transform: Callable[..., Any] = lambda x: x,
+        check_compute_fn: bool = True,
+        device: str | torch.device = torch.device("cpu"),
```

Comment:
Docs build is failed due to the pipe in the type annotation, let's use Union. We still support python 3.8 and 3.9 before their EOL.